newhelm/caching.py

```diff
@@ -70,31 +80,36 @@ def get_cached_response(self, request):
     def update_cache(self, request, response):
         if not self._can_encode(request) or not self._can_encode(response):
             return
-        encoded_request = self._encode_request(request)
+        cache_key = self._hash_request(request)
         encoded_response = self._encode_response(response)
-        self.cached_responses[encoded_request] = encoded_response
+        self.cached_responses[cache_key] = encoded_response
         self.cached_responses.commit()
 
-    def _load_cached_responses(self):
+    def _load_cached_responses(self) -> SqliteDict:
         os.makedirs(self.data_dir, exist_ok=True)
         path = os.path.join(self.data_dir, self.fname)
-        return SqliteDict(path)
+        cache = SqliteDict(path, tablename=self._CACHE_SCHEMA_VERSION)
+        tables = SqliteDict.get_tablenames(path)
```

Comment:
Ah, interesting! If you have multiple tables allowed, you could also have something like a metadata table. Or things like ActiveRecord and Alembic have tables to track migrations applied. But I think this is fine for now and you should ship it. We'll know more the first time we need to do a migration and we can figure it out then.